operations/helm/charts/mimir-distributed/values.yaml

```diff
@@ -134,6 +134,7 @@ mimir:
     blocks_storage:
       backend: s3
       bucket_store:
+        max_chunk_pool_bytes: 12884901888 # 12GiB
```

Comment:
Where is this allocated? Cause we don't request so much memory for store-gateway for example

Reply:
It seems like the best thing to do is to set this equal to the memory request on the store-gateway. This is what we do in Jsonnet (both are 12GB). In the Helm chart, we don't have a single value.

`values.yaml` uses `512M` as the default
`small.yaml` uses `1.5Gi`
`large.yaml` uses `6Gi`

Maybe a better approach would be to set it differently via the `structuredConfig` in each of those three files? What do you think?

I don't see any simple way to convert the current request to bytes unless I write a template to actually parse the units. 🤔

Reply:
The default if left unset is `2GiB`, so the current state is that the store-gateways can exceed their requests on both the default and `small.yml` configurations.

I think it would be neat if we could have Mimir detect the current resource limits and set this value correctly, but not sure if it's worth it right now.

Reply:
Actually, let me check some profiles from live cells. I have a feeling the chunk pool is much smaller than I think.

Reply:
Checked out ops cell and I can see allocations in the 10s of MB.
Checked one of our large multi-tenant cells in production and I see ~380MB allocated by the pool.
Checked one of our large single-tenant cells in production and I see ~6MB allocated. 

Based on this, I think the only real purpose for this setting is as a safety mechanism to prevent unnecessary OOM, but it honestly doesn't do that great of a job at that, since store-gateways allocate most of their memory in other places. What do you think?

Reply:
Based on this we can keep it to have better jsonnet-helm parity. Hopefully the SaaS team issue about better memory protection will deal with this, I'll make a comment there

Reply:
Agreed, thanks!